api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -142,26 +143,37 @@
                                       ["SELECT id AS offset FROM event_log WHERE id = ?" offset]))]
         (boolean (:offset result)))))
 
-(defn process-unilog-events [offset db instance-id remote-api]
+(defn read-event-log
+  [db offset]
+  (jdbc/reducible-query db
+                        ["SELECT id, payload::text AS payload FROM event_log WHERE id > ? ORDER BY id ASC LIMIT 300" offset]
+                        {:auto-commit? false :fetch-size 300}))
+
+(defn process-unilog-events [offset db instance-id remote-api email]
   (ds/with-remote-api remote-api instance-id
     (let [ds (DatastoreServiceFactory/getDatastoreService)
-          events (process-new-events
-                   (jdbc/reducible-query db
-                                         ["SELECT id, payload::text AS payload FROM event_log WHERE id > ? ORDER BY id ASC LIMIT 300" offset]
-                                         {:auto-commit? false :fetch-size 300}))
+          events (process-new-events (read-event-log db offset))
+          user-id (user/id-by-email-or-throw-error remote-api instance-id email)
+          authorized-forms (set (map ds/id (su/list-forms user-id)))
```

Comment:
Why not load just the required `(:forms-to-load events)`? I expect that in each batch there will be a little number of different forms to load.